src/Compilers/Core/Portable/Symbols/ITypeSymbol.cs

```diff
@@ -94,4 +94,14 @@ public interface ITypeSymbol : INamespaceOrTypeSymbol
         /// </param>
         ISymbol FindImplementationForInterfaceMember(ISymbol interfaceMember);
     }
+
+    static internal class ITypeSymbolHelpers
+    {
+        // Intentionally not an extension method. We don't want it ever be called for symbol classes
+        // Once Default Interface Implementations are supported, we can move this method into the interface. 
+        internal static bool IsNullableType(ITypeSymbol typeOpt)
```

Comment:
:thought_balloon: it's not clear to me why we still have three implementations of the same method, or why some of them handle null inputs while others don't.

Reply:
> it's not clear to me why we still have three implementations of the same method, or why some of them handle null inputs while others don't.

For classes we don't ever want to go through interface dispatch. And we don't expect compilers to call those helpers with null as an input.